File: majority_element/majority_element.py
#!/usr/bin/env python3


def majority_element(nums):
    maj_line = len(nums)//2
    tally = {}
    for x in nums:
        tally[x] = tally.get(x, 0) + 1
        if tally[x] > maj_line:
            return x

    return -1


if __name__ == '__main__':
    nums = [2,2,1,1,1,2,2]
    print(majority_element(nums))

File: majority_element/majority_element.go
package main

import (
	"fmt"
)

func majority_element(nums []int) int {
	majority := len(nums) / 2  // integer division in Go is always floored
	tally := map[int]int{}
	for _, num := range nums {
		tally[num] = tally[num] + 1
		if tally[num] > majority {
			return num
		}
	}
	fmt.Println(tally)
	return -1
}



func main() {
	nums := []int{2, 2 ,1 ,1 ,1 ,2 ,2 }
	result := majority_element(nums)
	fmt.Println(result)
}


